### PR TITLE
feat: Export data

### LINKED ISSUE #30: Enable exporting of data
Allow users to export the SBOM, vulnerability data, Vaadin analysis, and their own investigation results (e.g., to a CSV file).

### PR DESCRIPTION
**Pull request by sosa-vaadin**
## Description

Add a button to both the vulnerabilities tab and the dependencies tab to export the data shown in their respective grids.

I also introduced notifications and a logger for both the vulnerabilities tab and the dependencies tab.

### Added dependencies

- [Apache Commons CSV 1.10.0](https://mvnrepository.com/artifact/org.apache.commons/commons-csv/1.10.0)
- [Vaadin Notification](https://mvnrepository.com/artifact/com.vaadin/vaadin-notification-flow/24.2.4)

Fixes #30

### CODE REVIEW

File: appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java [hunk 1]
```diff
@@ -116,6 +125,38 @@ public void refresh() {
...
+                        dependency.getRiskScore()
+                );
+            }
+
+            String fileName = "dependencies.csv";
+            StreamResource streamResource = new StreamResource(fileName, () -> new ByteArrayInputStream(outputStream.toByteArray()));
+            updateExportData(streamResource);
+        } catch (IOException e) {
+            // TODO handle exception properly
+            e.printStackTrace();
```
Comment (sosa-vaadin) [inline appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java:158]: I am not sure what to do here.

File: appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesTab.java [hunk 2]
```diff
@@ -107,11 +116,44 @@ protected void applyFilters() {
...
+                        vulnerability.getDeveloperAnalysis()
+                );
+            }
+
+            String fileName = "vulnerabilities.csv";
+            StreamResource streamResource = new StreamResource(fileName, () -> new ByteArrayInputStream(outputStream.toByteArray()));
+            updateExportData(streamResource);
+        } catch (IOException e) {
+            // TODO handle exception properly
+            e.printStackTrace();
```
Comment (sosa-vaadin) [inline appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesTab.java:155]: I am not sure what to do here.

File: appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java (hunk 1 above)
Comment (tamasmak) [reply appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java:158]: You could throw an [AppSecException](https://github.com/vaadin/appsec-kit/blob/main/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecException.java#L20) with some message.

File: appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesTab.java (hunk 2 above)
Comment (tamasmak) [reply appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesTab.java:155]: Same here.

File: appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java (hunk 1 above)
Comment (tamasmak) [reply appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java:158]: I see. Then, what about showing an error log in dev tools' log tab and also logging an error or warn in the logs?

Comment (tamasmak) [reply appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java:158]: But maybe an exception would be more suitable for these kind of errors. We could create an `AppSecException` in starter module too or make the other one public. What do you think? (cc @heruan )

Comment (heruan) [reply appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesTab.java:158]: If there an error during the export, the application shouldn't throw. Let's catch the exception as it is and push a notification to the UI with the error message, to make it clear for the user that something went wrong.

**Comment by tamasmak** [comment]
Looks good! The ticket also mentions the export of the Vaadin and the own developer analysis. Do we want to make those exportable too?

**Comment by sosa-vaadin** [comment]
> Looks good! The ticket also mentions the export of the Vaadin and the own developer analysis. Do we want to make those exportable too?

They are included in the last two columns of the vulnerabilities export. @tarekoraby should check if the requirement is met as he envisioned.

**Comment by tarekoraby** [comment]
It looks good to me.

**Review by tamasmak — APPROVED** [review]

**Pull request merged**